app/controllers/lesson_plans_controller.rb

```diff
@@ -59,6 +70,22 @@ def create_lesson
     end
   end
 
+  def destroy_lessons_by_topic
+    topic = Topic.find(params[:topic_id])
+    @lesson_plan = current_lesson_plan
+    lessons = topic.lessons
+
+    lessons.each do |lesson|
+      @lesson_plan.planned_lessons.
```

Comment:
Non-blocking: `current_planned_lesson(lesson).destroy` should work here as well I think.